include/exadg/solvers_and_preconditioners/preconditioners/elementwise_preconditioners.h

```diff
@@ -148,6 +159,14 @@ class InverseMassPreconditioner
   {
     integrator = std::make_shared<Integrator>(matrix_free, dof_index, quad_index);
     inverse    = std::make_shared<CellwiseInverseMass>(*integrator);
+
+    dealii::FiniteElement<dim> const & fe = matrix_free.get_dof_handler(dof_index).get_fe();
+
+    AssertThrow(
+      fe.conforms(dealii::FiniteElementData<dim>::L2) and
+        (fe.base_element(0).dofs_per_cell == dealii::Utilities::pow(fe.degree + 1, dim)),
```

Comment:
@kronbichler If I understand your other comment correctly, we would have to assert here (where we consider an inverse mass preconditioner for the elementwise, local problem) that the number of quadrature points is equal to the number of nodes?

Reply:
Yes, exactly, otherwise the operator will not work. Or maybe it works, but it does not compute an appropriate inverse but some kind of pseudo inverse.

Reply:
Ok, then I need to find out how to ask `MatrixFree` for the number of 1D quadrature points given a `quad_index`.

Reply:
You should be able to ask
```c++
matrix_free.get_shape_info(0, quad_index).data[0].n_q_points_1d;
```

Reply:
thanks! already included in the newest version